service/history/workflow_task_handler_callbacks.go

```diff
@@ -606,6 +606,23 @@ func (handler *workflowTaskHandlerCallbacksImpl) handleWorkflowTaskCompleted(
 			return nil, err
 		}
 
+		if !ms.IsWorkflowExecutionRunning() {
```

Comment:
I suggest we extract this code into a separate method. This method is too long already.

Reply:
It is a call to a single function + error handling. This entire function must be split in may sub function. I will do it later.